src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellItemRenderer.cs

```diff
@@ -433,6 +435,18 @@ public override void ViewWillLayoutSubviews()
 			base.ViewWillLayoutSubviews();
 		}
 
+		async Task UpdateNavBarHidden(ShellSection shellSection)
+		{
+			if (SelectedViewController is UINavigationController navigationController && _displayedPage is not null)
+			{
+				// We need to wait for the layout to finish before we can set the navigation bar hidden,
+				// otherwise it will not animate correctly.
+				await Task.Yield();
```

Comment:
from your comment, I would say that `Task.Yield` isn't very reliable here. Because you aren't awating for something, the `Task.Yield` will just make sure your method will NOT complete asynchronously. You can find a more complete explanation [here](https://stackoverflow.com/a/23441833)